tests/src/unit/WebdamConfigFormTest.php

```diff
@@ -19,82 +17,125 @@
-//  public function testSubmitForm() {
-//    $config_stub = new FormConfigStub();
-//    $config_factory_stub = new FormConfigFactoryStub();
-//    $config_factory_stub->set('media_webdam.settings', $config_stub);
-//
-//    $wconfig = new WebdamConfig($config_factory_stub);
-//
-//    $form_state = new FormState();
-//    $form_state->set('username', 'webdam_username');
-//    $form_state->set('password', 'webdam_pw');
-//    $form_state->set('client_id', 'webdam_client_id');
-//    $form_state->set('secret', 'webdam_client_secret');
-//
-//    $form = [];
-//
-//    $wconfig->submitForm($form, $form_state);
-//
-//    $this->assertEquals('webdam_username', $config_stub->get('username'));
-//    $this->assertEquals('webdam_pw', $config_stub->get('password'));
-//    $this->assertEquals('webdam_client_id', $config_stub->get('client_id'));
-//    $this->assertEquals('webdam_client_secret', $config_stub->get('secret'));
-//  }
-
+  //  public function testSubmitForm() {
+  //    $config_stub = new FormConfigStub();
+  //    $config_factory_stub = new FormConfigFactoryStub();
+  //    $config_factory_stub->set('media_webdam.settings', $config_stub);
+  //
+  //    $wconfig = new WebdamConfig($config_factory_stub);
+  //
+  //    $form_state = new FormState();
+  //    $form_state->set('username', 'webdam_username');
+  //    $form_state->set('password', 'webdam_pw');
+  //    $form_state->set('client_id', 'webdam_client_id');
+  //    $form_state->set('secret', 'webdam_client_secret');
+  //
+  //    $form = [];
+  //
+  //    $wconfig->submitForm($form, $form_state);
+  //
+  //    $this->assertEquals('webdam_username', $config_stub->get('username'));
+  //    $this->assertEquals('webdam_pw', $config_stub->get('password'));
+  //    $this->assertEquals('webdam_client_id', $config_stub->get('client_id'));
+  //    $this->assertEquals('webdam_client_secret', $config_stub->get('secret'));
+  //  }
 }
 
 class FormConfigFactoryStub extends ConfigFactory {
   protected $configs = [];
   public function __construct() {}
-  public function get($name) { return $this->configs[$name]; }
-  public function getEditable($name) { return $this->configs[$name]; }
+  public function get($name) {
+    return $this->configs[$name];
+  }
+  public function getEditable($name) {
+    return $this->configs[$name];
+  }
   public function set($name, $config) {
     $this->configs[$name] = $config;
   }
+
 }
 
 class FormConfigStub extends Config {
   protected $data = [];
   public function __construct() {}
   public function save($has_trusted_data = FALSE) {}
-  public function set($key, $value) { $this->data[$key] = $value; }
-  public function get($key = '') { return $this->data[$key]; }
+  public function set($key, $value) {
+    $this->data[$key] = $value;
+  }
+  public function get($key = '') {
+    return $this->data[$key];
+  }
+
 }
 
+class WebdamTestStub implements WebdamInterface {
+  public function getSubscriptionDetails() {
```

Comment:
If you change the isAuthenticated() function as I mentioned above, this should return an object with the key that you're checking for.